### PR TITLE
docs: Headlamp on eks with cognito as oidc provider

### PR DESCRIPTION
**Pull request by ashu8912**

### CODE REVIEW

**Comment by illume** [comment]
Hey @ashu8912 nice work. I'll get to testing/reviewing it in the next days.

(ps. I made a few apps with Cognito back in the day, and now I'm thinking about them)

File: docs/installation/in-cluster/eks/_index.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+After setting up the aws cli, you can fetch the kubeconfig for the EKS cluster by running the following command
+```bash
+aws eks --region <YOUR_REGION_HERE> update-kubeconfig --name <YOUR_CLUSTER_NAME_HERE>
+```
+Where <YOUR_REGION_HERE> is the region where the EKS cluster is created and <YOUR_CLUSTER_NAME_HERE> is the name of the EKS cluster.
+
+## Deploying Headlamp on the EKS cluster
+1. First make sure you have helm installed on your local machine, If not you can follow the installation steps [here](https://helm.sh/docs/intro/install/)
+2. Once your cluster is operational, you need to configure a cluster role and a cluster role binding for the cognito user (You can create a cognito user from the aws console or through the sign up page which shows up when headlamp starts the login dance). This step is essential for enabling the Kubernetes API server to identify the user. In this example, we'll be associating the user with the predefined `cluster-admin` Role.
```
Comment (illume) [inline docs/installation/in-cluster/eks/_index.md:83]: I tried to make this a bit easier to read...

---

2. Your cluster should be running. 

3. Create a Cognito user through the AWS console. You will need the email for the next step.

4. Inside Kubernetes we need to give users a "cluster-admin" [ClusterRole](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole). We do this by setting up a ClusterRoleBinding. So the cluster can authorize these users.

... example


5. Create a `values.yaml` file...

File: docs/installation/in-cluster/eks/_index.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    oidc:
+    clientID: "<YOUR-CLIENT-ID>"
+    clientSecret: "<YOUR-CLIENT-SECRET>"
+    issuerURL: "https://cognito-idp.<YOUR_REGION_HERE>.amazonaws.com/<USER_POOL_ID>"
+    scopes: "profile,email,openid"
+```
+
+Replace `<YOUR-CLIENT-ID>`,`<YOUR-CLIENT-SECRET>`,`<YOUR_REGION_HERE`, `<USER_POOL_ID>` with your specific cognito user pool app configuration details.
+
+3. Save the `values.yaml` file and Install Headlamp using helm with the following commands:
```
Comment (illume) [inline docs/installation/in-cluster/eks/_index.md:120]: Check the numbering of the step here.

**Review by illume — CHANGES_REQUESTED** [review]
Hey hey.

I liked the tutorial, it seems to cover what's needed.

- Made some suggestions for your consideration. (I added some links to some, so maybe edit comment to easily copy the links if you want them.)
- I didn't run through the tutorial testing it yet.  (I can do that after you have addressed these suggestions so I have a better understanding)

File: docs/installation/in-cluster/eks/_index.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
+---
+title: How to Set Up Headlamp in EKS with cognito as the oidc provider
```
[the remark this answers is not in the document]
Comment (ashu8912) [reply docs/installation/in-cluster/eks/_index.md]: updated

File: docs/installation/in-cluster/eks/_index.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    - You can either use the hosted UI or create your own UI for the authentication workflows (We are using the Cognito hosted UI).
+    - You can also configure the domain name for the hosted UI (Also remember this is the issuer URL as well which we will later use to setup EKS with). We are using Cognito domain for this tutorial but you can chose your own custom domain as well.
+    - You can also configure the callback URLs for the auth flow. For this tutorial we are setting it to be 
+    http://localhost:8000/oidc-callback as we will be running headlamp on this port but you can chose your own callback host as well based on your needs make sure to append /oidc-callback to the host since this is where headlamp expect the redirect from auth to occur.
+    - Client Secret - We want to generate a client secret as our headlamp app needs it to start the auth dance, So select Generate a client secret client secret column.
+    - You can also configure the scopes and claims for the tokens. For this tutorial make sure openid, profile, email scopes are selected, other scopes are optional.
+
+![Integrate your app](./cognito/integrate-your-app1.png)
```
[the remark this answers is not in the document]
Comment (ashu8912) [reply docs/installation/in-cluster/eks/_index.md]: updated

File: docs/installation/in-cluster/eks/_index.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+8. Click next to go to the Select Add-ons view, For this tutorial we will use the default selected add-ons.
+![Select Add-ons](./cluster/select_addons.png)
+9. Click next to go to the Configure selected add-ons view, For this tutorial we will use the default settings.
+![Configure Add-ons](./cluster/configure_addons.png)
+10. Click next to go to the Review and Create view where you can review all the settings and create the cluster.
+
+## Configuring OIDC provider for EKS
+After creating the EKS cluster, we need to configure the OIDC provider for the cluster. This will allow the cluster to authenticate users using Cognito.
+1. Go to the access tab on the cluster overview page of the cluster you created above and click on the Associate OIDC Identity provider.
+2. In the Associate OIDC Identity provider view, Enter a name for the Identity Provider Configuration, enter the issuer URL which can be found in the cognito user pool view you created. It should be something like https://cognito-idp.<YOUR_REGION_HERE>.amazonaws.com/USER_POOL_ID, Client ID can be found in the app integration tab on the user pool overview page. The other fields can be filled as shown in the image below.
```
[the remark this answers is not in the document]
Comment (ashu8912) [reply docs/installation/in-cluster/eks/_index.md]: updated

File: docs/installation/in-cluster/eks/_index.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
+---
+title: How to Set Up Headlamp in EKS with cognito as the oidc provider
+linktitle: 'Tutorial: Headlamp on EKS with cognito'
+---
+
+In this tutorial we will walk through the process of configuring Headlamp within an EKS cluster while utilizing Cognito for OIDC (OpenID Connect) authentication. This tutorial is based on Cognito, EKS Platform version eks.6, Kubernetes version 1.29 and Headlamp version 0.23.2.
```
[the remark this answers is not in the document]
Comment (ashu8912) [reply docs/installation/in-cluster/eks/_index.md]: updated

File: docs/installation/in-cluster/eks/_index.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+## Configuring Cognito
+Cognito is a fully managed identity provider that allows you to create and manage users for your applications. In this tutorial, we will use Cognito as the OIDC provider for our EKS cluster. Before proceeding, follow the [getting started guide](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools.html) to know more about cognito.
+
+We will start by creating a cognito user pool, For this tutorial we will limit ourselves mostly to the AWS portal but you can use the ekctl or aws CLIs to configure.
```
[the remark this answers is not in the document]
Comment (ashu8912) [reply docs/installation/in-cluster/eks/_index.md]: updated

File: docs/installation/in-cluster/eks/_index.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+2. Signin to the aws cli can be done by following the steps [here](https://docs.aws.amazon.com/signin/latest/userguide/command-line-sign-in.html)
+
+After setting up the aws cli, you can fetch the kubeconfig for the EKS cluster by running the following command
+```bash
+aws eks --region <YOUR_REGION_HERE> update-kubeconfig --name <YOUR_CLUSTER_NAME_HERE>
+```
+Where <YOUR_REGION_HERE> is the region where the EKS cluster is created and <YOUR_CLUSTER_NAME_HERE> is the name of the EKS cluster.
+
+## Deploying Headlamp on the EKS cluster
+1. First make sure you have helm installed on your local machine, If not you can follow the installation steps [here](https://helm.sh/docs/intro/install/)
```
[the remark this answers is not in the document]
Comment (ashu8912) [reply docs/installation/in-cluster/eks/_index.md]: updated

File: docs/installation/in-cluster/eks/_index.md (hunk 1 above)
Comment (ashu8912) [reply docs/installation/in-cluster/eks/_index.md]: updated

File: docs/installation/in-cluster/eks/_index.md (hunk 2 above)
Comment (ashu8912) [reply docs/installation/in-cluster/eks/_index.md]: updated

**Review by illume — APPROVED** [review]
👍 thanks

**Pull request merged**